lib/rouge/lexers/supercollider.rb

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+module Rouge
+  module Lexers
+    class SuperCollider < RegexLexer
+      tag 'supercollider'
+      filenames '*.sc', '*.scd'
+
+      title "SuperCollider"
+      desc 'A cross-platform interpreted programming language for sound synthesis, algorithmic composition, and realtime performance'
+
+      keywords = Set.new %w(
+        var arg classvar const super this
+      )
+
+      # these aren't technically keywords, but we treat
+      # them as such because it makes things clearer 99%
+      # of the time
+      reserved = Set.new %w(
+        case do for forBy loop if while new newCopyArgs
+      )
+
+      constants = Set.new %w(
+        true false nil inf thisThread
+        thisMethod thisFunction thisProcess
+        thisFunctionDef currentEnvironment
+        topEnvironment
+      )
+
+      start { push :bol }
```

Comment:
Is there anything special that's perhaps at the beginning of the file?

You're using a `:bol` state which is usually what that's for but it looked to me like the actual content is just the ordinary comments that are captured by the rules you mix in at the beginning of the `:root` state.